include/dlaf/matrix/panel.h

```diff
@@ -387,6 +383,14 @@ struct Panel : public Panel<axis, const T, device> {
       return splitTile(tile, {{0, 0}, tileSize(index)});
   }
 
+  /// This functions sets all the elements of the tiles in the range to zero without inhibiting to set the
+  /// tile as external afterward (e.g. the tile is set to zero without changing its state to "already used")
+  template <class ExecutorOrPolicy>
+  friend void set0(ExecutorOrPolicy ex, Panel<axis, T, Device::CPU>& panel) {
```

Comment:
IMHO there is a thing that it is worth highlighting.

`set0` does not touch the status of the tile, i.e. it will still be possible to link external tiles, which is an explicit requirement for the use-case I have in reduction to band.
But this also means that `set0` will always ignore the status of the tile even if it has been previously set as external. Indeed, it just targets internal tiles in the range, and if one of the tile is set as external (so it is RO), it does not care and sets to 0 the internal one (that won't be used at all, since it is hidden by the external tile).

I think that for the moment it is ok to keep this functionality as simple as it is, and  just add this specification to the documentation. What do you think, @rasolca?

Reply:
Fine with me.

Reply:
As quickly discussed offline, we opted for making set0 more compliant with the Panel current constraints.

For this reason, it will just access all the tiles as a user would do it; it means that, since it access all the tiles of the currently active range in the panel, it will be able to work just if all tiles are RW, and after its call it won't be possible anymore to set tile as externals (until the next `reset`).